Documentation for dso.docker.com (#15540)
docker/docker.github.io: atomist/get-started.md

---
title: Getting started
description: Getting started with Atomist
keywords: atomist, software supply chain, vulnerability scanning, tutorial
toc_max: 2
---

{% include atomist/disclaimer.md %}

To get started with Atomist, you'll need to:

> - Connect Atomist with your container registry

Author: ACK, already mention this in the Docker Hub section, but will consider whether to remove that section and mention it here instead. Although will probably keep it for parallelism.

<!-- GitHub Container Registry -->

To integrate Atomist with GitHub Container Registry, connect your GitHub
account, and enter a personal access token for Atomist to use when pulling
container images.

1. Go to <https://dso.docker.com> and sign in using your Docker ID credentials.
2. Connect your GitHub account as instructed in the
   [GitHub app page](./integrate/github.md#connect-to-github). Install the app
   into the GitHub account that contains your GitHub Container Registry.
3. Open the [**Integrations**](https://dso.docker.com/r/auth/integrations){:
   target="blank" rel="noopener" class=""} tab, and click the **Configure** link
   next to the **GitHub Container Registry** in the list of integrations.
4. Fill out the fields and click **Save Configuration**.

   The **Personal access token** is required for scanning images in private
   repositories. The token must have the
   [`read:packages` scope](https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages).

   You can leave the **Personal access token** field blank if you only want to
   scan images in public repositories.

</div>
<hr>
</div>

> ## Link images to Git repository

Reviewer: We should probably document the opt-in in buildx here as well. See https://github.com/docker/buildx/issues/1290

Knowing the source repository of an image is a prerequisite for Atomist to
interact with the Git repository, such as adding scan results as a status check
in pull requests. For Atomist to be able to link scanned images to a Git
repository repository, the image must be annotated with labels at build time.

The image labels that Atomist requires are:

| Label                                | Value                                             |
| ------------------------------------ | ------------------------------------------------- |
| `org.opencontainers.image.revision`  | The commit revision that the image is built for.  |

> | `org.opencountainers.image.source`   | HTTP(S) address of the project.                   |

Reviewer 2: Typo

```suggestion
| `org.opencontainers.image.source`   | HTTP(S) address of the project.                   |
```
About description, we should keep the same as OCI ones: https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys

```
| `org.opencontainers.image.revision` | Source control revision identifier for the packaged software. |
| `org.opencontainers.image.source`   | URL to get source code for building the image.                   |
```

And maybe also add a link to the specs so users are not confused: https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys

---
title: Get started
description: Getting started with Atomist
keywords: atomist, software supply chain, vulnerability scanning, tutorial
toc_max: 2
---

{% include atomist/disclaimer.md %}

To get started with Atomist, you'll need to:

- Connect Atomist with your container registry
- Link your container images with their Git source

> Before you can begin the setup, you’ll need a Docker ID. If you don't already

Reviewer 3: ```suggestion
Before you can begin the setup, you need a Docker ID. If you don't already
```

have one, you can [register here](https://hub.docker.com/signup){:
target="blank" rel="noopener" class=""}.

## Connect container registry

> After completing this setup, Atomist will have read-only access to your

Reviewer 3: I'd move this sentence to the top and make it part of the relevant bullet point. Having it here confused me for a second, it felt like I'd skipped something.

Author: Good point... I tried to change the introduction text for this section to clarify. I think it could be good to have this information up-front, about what access Atomist requires and why.

Atomist, the following AWS resources are required:

- Read-only Identity Access Management (IAM) role, for Atomist to be able to
  access the container registry
- Amazon EventBridge, to notify Atomist of pushed and deleted images

This procedure uses pre-defined CloudFormation templates to create the necessary
IAM role and Amazon EventBridge. This template protects you from
[confused deputy attacks](https://docs.aws.amazon.com/IAM/latest/UserGuide/confused-deputy.html){:
target="blank" rel="noopener" class=""} by ensuring a unique `ExternalId`, along
with the appropriate condition on the IAM role statement.

1. Go to <https://dso.docker.com> and sign in using your Docker ID credentials.
2. Navigate to the **Integrations** tab and select **Configure** next to the
   **Elastic Container Registry** integration.
3. Fill out all the fields, except **Trusted Role ARN**. The trusted role
   identity is known only after applying the CloudFormation template.

   Choose basic auth credentials to protect the endpoint that AWS will use to
   notify Atomist. The URL and the basic auth credentials are parameters to the
   CloudFormation template.

> 4. Now create the CloudFormation stack. Before creating the stack, AWS will ask

Reviewer 3: ```suggestion
4. Now create the CloudFormation stack. Before creating the stack, AWS asks
```

             <img alt="Launch Stack" src="https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png" />
           </a>
         </td>
       </tr>
       <tr>
         <th>ap-southeast-2</th>
         <td>
           <a href="https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-2#/stacks/new?stackName=atomist-public-templates-ecr-integration&templateURL=https://s3.amazonaws.com/atomist-ap-southeast-2/atomist-public-templates/latest/ecr-integration.template">
             <img alt="Launch Stack" src="https://s3.amazonaws.com/cloudformation-examples/cloudformation-launch-stack.png" />
           </a>
         </td>
       </tr>
     </table>
   </div>

5. After creating the stack, copy the **Value** for the **AssumeRoleArn** key
   from the **Outputs** tab in AWS.

   ![AWS stack creation output](./images/ecr/stack-output.png){: width="700px"}

6. Paste the copied **AssumeRoleArn** value into the **Trusted Role ARN** field
   on the Atomist configuration page.

7. Select **Save Configuration**.

>    Atomist will now test the connection with your ECR registry. You'll see a

Reviewer 3: ```suggestion
   Atomist tests the connection with your ECR registry.  A
```

>    green check mark beside the integration if a successful connection was made.

Reviewer 3: ```suggestion
   green check mark displays beside the integration if a successful connection is made.
```

</div>
<hr>
</div>

## Link images to Git repository

Knowing the source repository of an image is a prerequisite for Atomist to
interact with the Git repository. For Atomist to be able to link scanned images
back to a Git repository repository, you must annotate the image at build time.

The image labels that Atomist requires are:

| Label                                | Value                                             |
| ------------------------------------ | ------------------------------------------------- |
| `org.opencontainers.image.revision`  | The commit revision that the image is built for.  |
| `com.docker.image.source.entrypoint` | Path to the Dockerfile, relative to project root. |

For more information about pre-defined OCI annotations, see the
[specification document on GitHub](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys).

You can add these labels to images using the built-in Git provenance feature of
Buildx, or set using the `--label` CLI argument.

### Add labels using Docker Buildx

> Beta
>

> > Git provenance labels via Buildx is a beta feature.

Reviewer 3: Link to the product release lifecycle page?
